catroid/src/main/java/org/catrobat/catroid/embroidery/JumpStitchPoint.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.embroidery;
+
+import android.graphics.PointF;
+
+import org.catrobat.catroid.content.Sprite;
+
+public class JumpStitchPoint implements StitchPoint {
```

Comment:
The intent was, that Stitchpoints actually had a purpose apart from being containers. If you let the points act on some kind of workspace this should simplify things.
Regarding color and layer changes: Can you explain to me how this algorithm should actually work?
It seems to me that having a stack/array of layers and letting the stich points act on those layers might simplify the logic here even more. (Maybe I am wrong here :/ )